addons/website_sale/controllers/combo_configurator.py

```diff
@@ -81,3 +82,16 @@ def website_sale_combo_configurator_update_cart(
         request.session['website_sale_cart_quantity'] = order_sudo.cart_quantity
 
         return values
+
+    def _get_combo_item_data(
+        self, combo, combo_item, selected_combo_item, date, currency, pricelist, **kwargs
+    ):
+        data = super()._get_combo_item_data(
+            combo, combo_item, selected_combo_item, date, currency, pricelist, **kwargs
+        )
+        # If the root product of type 'combo' is published, but the combo_item linked to it,
+        # the image of the item will not be accessible to public/portal users.
+        # Therefore, we send the raw image from here
+        if not combo_item.product_id.is_published and combo_item.product_id.image_128:
+            data['product']['image_src'] = image_data_uri(combo_item.product_id.image_128)
```

Comment:
you have to somehow add this only to frontend request, because right now, it's also sent (and used) when opening the combo configurator in the backend :eyes:

Reply:
Indeed :man_facepalming: (ok for the cart notification then :D)